mne/viz/utils.py

```diff
@@ -1691,9 +1696,17 @@ def _compute_scalings(scalings, inst):
                                  % (key, value))
             continue
         this_data = data[ch_types[key]]
-        scale_factor = np.percentile(this_data.ravel(), [0.5, 99.5])
-        scale_factor = np.max(np.abs(scale_factor))
-        scalings[key] = scale_factor
+        if remove_dc and (this_data.shape[1] / inst.info["sfreq"] >= 10):
+            length = int(10 * inst.info["sfreq"]) # length of segments (in s)
```

Comment:
similarly this should be `duration` not `10`

Reply:
If the segment length is `duration` I probably don't have to check for the length of the visible data at all.